custom_metrics/wpt_bodies.js

```diff
@@ -0,0 +1,1366 @@
+              else {
+                target.text_links++;
+              }
+
+              if (crawlable) { // unless nofollow ???
+                if (follow) {
+                  target.crawlable.follow++;
+                } else
+                {
+                  target.crawlable.nofollow++;
+                }
+              }
+
+            });
+          }
+          return target;
+        };
+
+        let result = {};
+
+        result.rendered = getAnchorData(document)
+
+        let rawHtmlDocument = getRawHtmlDocument();
+
+        if (rawHtmlDocument) {
+          result.raw = getAnchorData(rawHtmlDocument);
+        }
+
+        return result;
+      }
+      catch(e) {
+        return logError("anchors", e);
+      }
+    })(),
+
+
+    // Extract the real title tag contents  
+    // Used by: SEO
+    // Backup: old version in almanac.js 
+    'title': (() => {
+      try {    
+        let result = {};
+
+        function getTitles(d) {
+          let target = {};
+          target.total = Array.from(d.querySelectorAll('head title')).map(e => {
+            let text = e.innerText.trim();
+            let characters = text.length;
+            let words = text.match(/\S+/g)?.length;
+
+            if (text.length > 0 && !target.primary) {
+              if (d == document) {
+                renderedPrimaryTitle = text; // for the heading section
+              }
+              else {
+                rawPrimaryTitle = text; // for the heading section
+              }
+
+              let snippet = text;
+
+              if (snippet.length > 200) {
+                snippet = snippet.substring(0,200) + "...";
+              }
+
+              target.primary = {
+                characters: characters,
+                words: words,
+                text: snippet
+              };
+            }
+            return {characters, words };
+          }).length;
+
+          return target;
+        }
+
+        result.rendered = getTitles(document);
+
+        let rawHtmlDocument = getRawHtmlDocument();
+
+        if (rawHtmlDocument) {
+          result.raw = getTitles(rawHtmlDocument);
+        }
+
+        result.title_changed_on_render = renderedPrimaryTitle != rawPrimaryTitle;
+
+        return result; 
+      }
+      catch(e) {
+        return logError("title", e);
+      }
+    })(),
+
+    // Extract the real meta description tag contents  
+    // Used by: SEO
+    // Backup: meta nodes in almanac.js 
+    'meta_description': (() => {
+      try {    
+        let result = {};
+
+        function getMetaDescriptions(d) {
+          let target = {all: {text: "", words: 0, characters: 0}};       
+          target.total = Array.from(d.querySelectorAll('head meta[name="description"]')).map(e => {
+            let text = e.getAttribute("content") ?? "";
+            let characters = text.length;
+            let words = text.match(/\S+/g)?.length;
+
+            target.all.text = (target.all.text+" "+text).trim();
```

Comment:
Nit:
```suggestion
            target.all.text += ` ${text.trim()}`;
```

Reply:
Ah ok, updated my suggestion. It's not the `text` var that you necessarily want to trim, but the concatenated value.

Reply:
```
let test = "previous text";
let text = " new text ";
test += ` ${text}`.trim();
console.log(test); // result = 'previous textnew text'
```

Reply:
It's now:
target.all.text = (target.all.text+" "+text.trim()).trim();